packages/js/src/executors/tsc/lib/get-in-memory-tsconfig.ts

```diff
@@ -0,0 +1,55 @@
+import type { ExecutorContext } from '@nx/devkit';
+import { joinPathFragments, readJsonFile } from '@nx/devkit';
+import * as ts from 'typescript';
+import type { NormalizedExecutorOptions } from '../../../utils/schema';
+import { getTaskOptions } from './batch/get-task-options';
+import type { TypescriptInMemoryTsConfig } from './typescript-compilation';
+
+export function getInMemoryTsConfigWithProjectReferences(
+  taskOptionsMap: Record<string, NormalizedExecutorOptions>,
+  taskName: string,
+  taskOptions: NormalizedExecutorOptions,
+  context: ExecutorContext
+): TypescriptInMemoryTsConfig {
+  const originalTsConfigPath = taskOptions.tsConfig;
+
+  const projectReferences: { path: string }[] = [];
+  for (const depTask of context.taskGraph.dependencies[taskName]) {
+    // if included in the provided map, use it
+    if (taskOptionsMap[depTask]) {
```

Comment:
There is an edge case here if the dep doesn't have `tsConfig` option. It shouldn't come up if all deps are using tsc executor, but it is technically possible. Maybe a check on `if (taskOptionsMap[depTask]?.tsConfig)`?

And then at the end of the `for` block, throw and error if no `tsConfig` is found for the dep (either from task or cache).

Reply:
Not really. At least not from the intended and current usage. The `taskOptionsMap` object contains only the options of tasks that are part of the batch execution, so all those tasks have the same executor (tsc). The `tsConfig` option is required for that executor, so if not provided, the schema validation should throw before getting here.

Dependencies with a different executor are executed before and are not part of the batch execution.